spring-boot-rest/src/test/java/com/baeldung/common/web/AbstractBasicLiveTest.java

```diff
@@ -0,0 +1,103 @@
+package com.baeldung.common.web;
+
+import static com.baeldung.web.util.HTTPLinkHeaderUtil.extractURIByRel;
+import static org.apache.commons.lang3.RandomStringUtils.randomNumeric;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+
+import java.io.Serializable;
+import java.util.List;
+
+import org.junit.Test;
+
+import com.google.common.net.HttpHeaders;
+
+import io.restassured.RestAssured;
+import io.restassured.response.Response;
+
+public abstract class AbstractBasicLiveTest<T extends Serializable> extends AbstractLiveTest<T> {
+
+    public AbstractBasicLiveTest(final Class<T> clazzToSet) {
+        super(clazzToSet);
+    }
+
+    // find - all - paginated
+
+    @Test
+    public void whenResourcesAreRetrievedPaged_then200IsReceived() {
+        create();
+        
+        final Response response = RestAssured.get(getURL() + "?page=0&size=10");
+
+        assertThat(response.getStatusCode(), is(200));
+    }
+
+    @Test
+    public void whenPageOfResourcesAreRetrievedOutOfBounds_then404IsReceived() {
+        final String url = getURL() + "?page=" + randomNumeric(5) + "&size=10";
+        final Response response = RestAssured.get(url);
+
+        assertThat(response.getStatusCode(), is(404));
+    }
+
+    @Test
+    public void givenResourcesExist_whenFirstPageIsRetrieved_thenPageContainsResources() {
+        create();
+
+        final Response response = RestAssured.get(getURL() + "?page=0&size=10");
+
+        assertFalse(response.body().as(List.class).isEmpty());
+    }
+
+    @Test
+    public void whenFirstPageOfResourcesAreRetrieved_thenSecondPageIsNext() {
+        create();
+        create();
+        create();
```

Comment:
Some of these tests changed a little bit, creating o couple of entities before making the assertions.
In the old module, I think persistence tests were carried out as part of the test suite before running these tests, therefore they didn't need to be created explicitly. Now each live test should run successfully even if they are run alone.